plugins/statistics.koplugin/main.lua

```diff
@@ -234,28 +291,112 @@ function getDatesForBookOldFormat(book)
     return generateReadBooksTable(book.title, dates)
 end
 
-function getDatesForBook(book)
+-- sdays -> number of days to show
+-- ptype -> daily - show daily without weekday name
+--          daily_weekday - show daily with weekday name
+--          weekly - show weekly
+--          monthly - show monthly
+function getDatesFromAll(sdays, ptype)
     local dates = {}
+    local sorted_performance_in_pages
+    local diff
+    local book = {}
+    now_t = os.date("*t")
+    local from_begin_day = now_t.hour *3600 + now_t.min*60 + now_t.sec
+    local now_stamp = os.time()
+    local one_day = 24 * 3600 -- one day in seconds
+    local period = now_stamp - ((sdays -1) * one_day) - from_begin_day
+    for _, v in pairs(ReadHistory.hist) do
+        local book_settings = DocSettings:open(v.file)
+        if book_settings.data.stats ~=nil then
+            book = book_settings.data.stats
+            --zeros table sorted_performance_in_pages
```

Comment:
Follow up on my previous review. You can do the following to save one table lookup:

Change

```lua
local book_settings = DocSettings:open(v.file)
if book_settings.data.stats ~=nil then
    book = book_settings.data.stats
```

To

```lua
local book_stats = DocSettings:open(v.file):readSetting('stats')
if book_stats ~= nil then
```

This way, you can also eliminate the function global `book` variable.